sdk/storage/azure_storage_blob/src/pipeline/storage_headers_policy.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use async_trait::async_trait;
+use azure_core::{headers::CLIENT_REQUEST_ID, Context, Policy, PolicyResult, Request};
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, Clone)]
+pub struct StorageHeadersPolicy;
+
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+impl Policy for StorageHeadersPolicy {
+    async fn send(
+        &self,
+        ctx: &Context,
+        request: &mut Request,
+        next: &[Arc<dyn Policy>],
+    ) -> PolicyResult {
+        if request
+            .headers()
+            .get_optional_string(&CLIENT_REQUEST_ID)
```

Comment:
This is fine for now, but we should define an `x-ms-client-request-id` policy in `azure_core` like most other languages have. I'll open a separate issue on core for this.